analytics/analytics/models.py

```diff
@@ -14,6 +14,15 @@ class Job(models.Model):
 
     # Derived fields
     package_name = models.CharField(max_length=128)
+    # these fields can be null because the data was not previously collected
+    package_version = models.CharField(max_length=128, null=True)
+    compiler_name = models.CharField(max_length=128, null=True)
+    compiler_version = models.CharField(max_length=128, null=True)
+    arch = models.CharField(max_length=128, null=True)
+    package_variants = models.TextField(null=True)
```

Comment:
What's the reason for making this a text field? Does it just have a truly arbitrary length?